src/Tribe/Commerce/PayPal/Endpoints.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/**
+ * Class Tribe__Tickets__Commerce__PayPal__Endpoints
+ *
+ * @since TBD
+ */
+class Tribe__Tickets__Commerce__PayPal__Endpoints {
+
+	/**
+	 * @var array
+	 */
+	protected $template_data = array();
+
+	/**
+	 * Hooks the class to actions and filters.
+	 *
+	 * @since TBD
+	 */
+	public function hook() {
+		add_filter( 'query_vars', array( $this, 'filter_query_vars' ) );
+		add_action( 'pre_get_posts', array( $this, 'replace_posts' ) );
+		add_filter( 'the_content', array( $this, 'filter_the_content' ) );
+		add_action( 'init', array( $this, 'add_rewrite_rules' ) );
+	}
+
+	public function add_rewrite_rules() {
+		$success_slug = preg_quote( $this->success_slug(), '/' );
+
+		// match `/tpp-success/WR45SDF5689FGD`
+		add_rewrite_rule(
```

Comment:
Shouldn't we use the `WP_Rewrite` class?

Reply:
We can, any particular reason why?
The `add_rewrite_rule` function is 2 lines of code and the code would not really change much:

```php
public function add_rewrite_rules() {
	global $wp_rewrite;

	$success_slug = preg_quote( $this->success_slug(), '/' );

	// match `/tpp-success/WR45SDF5689FGD`
	$wp_rewrite->add_rule(
		'^' . $success_slug . '/(\\w+)/?',
		'index.php?tribe-tpp-page=success&tribe-tpp-order=$matches[1]',
		'top'
	);
}
```